samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -113,14 +114,30 @@ def _generate_from_custom_location(
         """
         pipeline_template_git_location: str = click.prompt("Template Git location")
         if os.path.exists(pipeline_template_git_location):
-            return self._generate_from_pipeline_template(Path(pipeline_template_git_location))
+            pipeline_template_local_dir = Path(pipeline_template_git_location)
+        else:
+            with osutils.mkdir_temp(ignore_errors=True) as tempdir:
+                tempdir_path = Path(tempdir)
+                pipeline_template_local_dir = _clone_pipeline_templates(
+                    pipeline_template_git_location, tempdir_path, CUSTOM_PIPELINE_TEMPLATE_REPO_LOCAL_NAME
+                )
 
-        with osutils.mkdir_temp(ignore_errors=True) as tempdir:
-            tempdir_path = Path(tempdir)
-            pipeline_template_local_dir: Path = _clone_pipeline_templates(
-                pipeline_template_git_location, tempdir_path, CUSTOM_PIPELINE_TEMPLATE_REPO_LOCAL_NAME
+        try:
+            pipeline_templates_manifest: PipelineTemplatesManifest = _read_app_pipeline_templates_manifest(
+                pipeline_template_local_dir
+            )
+            # The manifest contains multiple pipeline-templates so select one
+            selected_pipeline_template_metadata: PipelineTemplateMetadata = _prompt_pipeline_template(
+                pipeline_templates_manifest
             )
-            return self._generate_from_pipeline_template(pipeline_template_local_dir)
+            selected_pipeline_template_dir: Path = pipeline_template_local_dir.joinpath(
+                selected_pipeline_template_metadata.location
+            )
+        except AppPipelineTemplateManifestException:
+            # If the repository does not contain a manifest, treat it as a pipeline template directory.
```

Comment:
This exception could be caused by a broken manifest too:

https://github.com/kfr2/aws-sam-cli/blob/e6b1faa57b234ad47ce61f1ef24a39fa7f996e63/samcli/commands/pipeline/init/pipeline_templates_manifest.py#L56

Maybe we can do a file existence check instead of using try-catch as a control flow.

Reply:
Thanks for letting me know! I've updated this pull request accordingly.